### DISCUSSION TITLE
Edit user does not work when there is only the installation Admin

### DISCUSSION THREAD

**Discussion by evertton** [nilsteampassnet/TeamPass#4382] [General]
When installing TeamPass, you will be asked to enter a password to create an admin account. By default, this account does not have an email address. The user will use this account only for administration/maintenance purposes and will not enter an email address.

Trying to edit a user, having only that administration account, without the email filled in, the edit fails without any feedback to the user. There is a feedback message, but it does not displayed, because the page is refreshed.

Here is the code snippet that reflects the question below: https://github.com/nilsteampassnet/TeamPass/blob/5a0744f8bf3e926411fae05a214f1bd1e5d47fdc/sources/users.queries.php#L1446C13-L1467C20

Which of the following is the best solution or do you suggest something else?

1. **Since email is a required field for creating users, request that the email field be filled in for Admin user setup on the installation screen.**
2. Inform the user that before being able to make editions, it is required to fill in the email field for the initial Admin user or create other administrator user.
3. Use any email in the admin setup to avoid the problem. (e.g. admin@teampass.teampass)

I highlighted solution 1, because it is my favorite. As for solution 3, it is quite elegant.

**Comment by nilsteampassnet** [comment]
You are right.
Solution 1 is the best one according me too.
